extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -537,6 +540,13 @@ static <DATA_TYPE> void writeColumnSource(
                         maxValuesPerPage,
                         maxOriginalRowsPerPage,
                         pageCount);
+                if (!usedDictionary) {
```

Comment:
Just hanging this here, since I can't comment on code you didn't edit. In `tryEncodeDictionary`, there's an early return for "all nulls".
```
            if (keyCount == 0 && hasNulls) {
                return false;
            }
```
I think that could be moved up to above the block that starts with 
```
            List<IntBuffer> arraySizeBuffers = null;
            if (lengthSource != null) {
```